var/spack/repos/builtin/packages/hpctoolkit/package.py

```diff
@@ -139,3 +175,18 @@ def configure_args(self):
             args.append('--enable-all-static')
 
         return args
+
+    # Remove setenv of ROCM, HIP, etc from the module file.  Loading
+    # the hpctoolkit module is not relevant to building a GPU app and
+    # some variables (HIP_PATH) intefere with building the app.
+    def setup_run_environment(self, env):
+        keeplist = []
+        for elt in env.env_modifications:
+            if not (isinstance(elt, SetEnv)
+                    and (elt.name.find('ROCM') >= 0
```

Comment:
`"ROCM" not in elt.name` to be more pythonic?

Reply:
we can leave that as it is to avoid conflict with upstream?